src/Test/DefaultTest.php

```diff
@@ -90,7 +86,39 @@ private function issetField(mixed $data, array $fields): bool
             return false;
         }
 
+        // Make sure that the key value is an array of strings
+        if (!is_array($data[$key]) || !$this->isListOfStrings($data[$key])) {
+            throw new InvalidArgumentException("Data structure is not as expected.");
+        }
+
         // Recursively check the next level
         return $this->issetField($data[$key], $fields);
     }
+
+    /**
+     * Checks if an array contains a list of strings.
+     *
+     * @param iterable<string> $value
+     *     The array to check.
+     *
+     * @return bool
+     *     Returns `true` if the array contains a list of strings, `false` otherwise.
+     *
+     * @throws InvalidArgumentException
+     *     If the input value is not an array.
+     */
+    private function isListOfStrings(iterable $value): bool
```

Comment:
IMHO, this name is misleading. If you are okay with any kind of list of strings, then there is no point checking for `is_array` below. But if you want to ensure that it _must_ be an array, then this function name is misleading. I suggest the function name: `isArrayOfStrings` in that case. It tells someone that this function will check both that it is an array (not just any iterable) and it only contains strings.

Reply:
@hussainweb Agreed! Addressed the same in this [commit](https://github.com/sadeesh-sdet/mdtt/commit/5a5b5adf3f3a732497d07ac232d10ec76a754d87)